omod/src/main/java/org/openmrs/module/appui/fragment/controller/SessionFragmentController.java

```diff
@@ -10,11 +11,18 @@
 import org.openmrs.ui.framework.fragment.action.ObjectResult;
 import org.springframework.web.bind.annotation.RequestParam;
 
+import javax.servlet.ServletContext;
+import javax.servlet.http.HttpServletResponse;
+
 public class SessionFragmentController {
 	
 	public FragmentActionResult setLocation(@RequestParam("locationId") Location location,
-	                   @SpringBean("locationService") LocationService locationService, UiSessionContext context) {
+											@SpringBean("locationService") LocationService locationService,
+											UiSessionContext context, ServletContext servletContext, HttpServletResponse httpResponse) {
 		context.setSessionLocation(location);
+		httpResponse.setHeader("Set-Cookie", AppUiConstants.COOKIE_NAME_LAST_SESSION_LOCATION + "=" + location.getLocationId()
```

Comment:
Nope. I mean if we want to set flags such as the HttpOnly we have to use the setHeader appraoch in servlet api version 2.5.